luaui/Include/blueprint_substitution/reports.lua

```diff
@@ -0,0 +1,182 @@
+            local extractsMetal = unitDef.extractsMetal or 0 
+            local energyMake = unitDef.energyMake or 0
+            local radarDist = unitDef.radarDistance or 0
+            local sonarDist = unitDef.sonarDistance or 0
+            local sightDist = unitDef.losRadius or 0
+            local canFly = false -- Buildings don't fly
+
+            local equivArm = equivalents[SIDES.ARM] or ""
+            local equivCor = equivalents[SIDES.CORE] or ""
+            local equivLeg = equivalents[SIDES.LEGION] or ""
+
+            local reportLine = table.concat({
+                escapeCsvField(categoryName), escapeCsvField(translatedHumanName), 
+                escapeCsvField(side or "???"), escapeCsvField(tier), escapeCsvField(isBuilding),
+                escapeCsvField(name), 
+                metalCost, energyCost, buildTime, 
+                health, speed, escapeCsvField(hasWeapons), 
+                escapeCsvField(hasBuildOptions), extractsMetal, energyMake,
+                radarDist, sonarDist, sightDist, escapeCsvField(canFly), 
+                escapeCsvField(tooltip),
+                escapeCsvField(equivArm), escapeCsvField(equivCor), escapeCsvField(equivLeg)
+            }, ",")
+            
+            Spring.Echo(reportLine)
+
+            -- Side totals logic remains the same, using 'side' derived earlier
+            if side == SIDES.ARM or side == SIDES.CORE or side == SIDES.LEGION then
+                sideTotals[side] = (sideTotals[side] or 0) + 1
+                if categoryName ~= "Misc" then
+                    sideComplete[side] = (sideComplete[side] or 0) + 1
+                else
+                    table.insert(uncategorizedUnits[side], reportLine) 
+                end
+            else
+                table.insert(uncategorizedUnits["UNKNOWN"], reportLine) 
+            end
+        else
+             Spring.Log("BlueprintReports", "warning", "[Mapping Report] Could not find UnitDef for ID: " .. tostring(buildingCoreData.unitDefID) .. " Name: " .. unitNameLower); 
+             -- No action needed here, loop continues automatically
+        end
+    end
+
+    Spring.Log("BlueprintReports", "info", "\n--- Mapping Report Summary (Buildings Only) ---")
+    for side, total in pairs(sideTotals) do
+        local complete = sideComplete[side] or 0
+        local percentage = total > 0 and (complete / total * 100) or 0
+        Spring.Log("BlueprintReports", "info", string.format("[Mapping Report] %s: %d / %d buildings categorized (%.2f%%)", side, complete, total, percentage))
+    end
+    local unknownCount = #uncategorizedUnits["UNKNOWN"]
+    if unknownCount > 0 then
+        Spring.Log("BlueprintReports", "info", string.format("[Mapping Report] Buildings with UNKNOWN side: %d", unknownCount))
+    end
+
+    Spring.Log("BlueprintReports", "info", "--- Blueprint Mapping Report End ---") 
+end
+
+function M.generateCategoryListReport()
+    Spring.Echo("\n--- Blueprint Category List Start ---")
```

Comment:
How are people supposed to use this? IIRC the console only shows like 2 lines tops so a big dump is probably not very readable. If they are supposed to read this from infolog then consider building a giant string (perhaps via `table.concat`) and doing a single Echo, because that way the engine won't append its `[t=0.0123456][frame=123123]` garbage before each line which makes it difficult to work with the output.

Reply:
> How are people supposed to use this?
`/blueprintcategorylist` and `/blueprintmapreport`

Will do.

Reply:
>> How are people supposed to use this?
>
> `/blueprintcategorylist` and `/blueprintmapreport`

I meant more in the sense of, when do they call one of the above (rarely, just to verify things, meaning it doesn't need to be super fancy? or more commonly?) and what do they do with its output (just a human glance, or does it make sense to make it machine parse friendly?).